libtracepoint-control-cpp/src/TracingPath.cpp

```diff
@@ -0,0 +1,247 @@
+        if (mountsFile != nullptr)
+        {
+            for (;;)
+            {
+                char line[4097];
+                if (!fgets(line, sizeof(line), mountsFile))
+                {
+                    break;
+                }
+
+                // line is "deviceName mountPoint fileSystem otherStuff..."
+
+                size_t linePos = 0;
+
+                // deviceName
+                while (IsNonSpaceChar(line[linePos]))
+                {
+                    linePos += 1;
+                }
+
+                // whitespace
+                while (IsSpaceChar(line[linePos]))
+                {
+                    linePos += 1;
+                }
+
+                // mountPoint
+                auto const mountPointBegin = linePos;
+                while (IsNonSpaceChar(line[linePos]))
+                {
+                    linePos += 1;
+                }
+                auto const mountPointEnd = linePos;
+
+                // whitespace
+                while (IsSpaceChar(line[linePos]))
+                {
+                    linePos += 1;
+                }
+
+                // fileSystem
+                auto const fileSystemBegin = linePos;
+                while (IsNonSpaceChar(line[linePos]))
+                {
+                    linePos += 1;
+                }
+                auto const fileSystemEnd = linePos;
+
+                if (!IsSpaceChar(line[linePos]))
+                {
+                    // Ignore line if no whitespace after fileSystem.
+                    continue;
+                }
+
+                std::string_view const fileSystem(line + fileSystemBegin, fileSystemEnd - fileSystemBegin);
+                std::string_view pathSuffix;
+                if (fileSystem == "tracefs"sv)
+                {
+                    // "tracefsMountPoint"
+                    pathSuffix = ""sv;
+                }
+                else if (fileSystem == "debugfs"sv)
+                {
+                    // "debugfsMountPoint/tracing"
+                    pathSuffix = "/tracing"sv;
+                }
+                else
+                {
+                    continue;
+                }
+
+                auto const mountPointLen = mountPointEnd - mountPointBegin;
+                auto const pathLen = mountPointLen + pathSuffix.size() + 1; // includes NUL
+                if (pathLen > sizeof(line))
+                {
+                    continue;
+                }
+
+                // path = mountpoint + suffix, e.g. "/sys/kernel/tracing\0"
+                memcpy(staticTracingDirBuffer, line + mountPointBegin, mountPointLen);
+                memcpy(staticTracingDirBuffer + mountPointLen, pathSuffix.data(), pathSuffix.size() + 1); // includes NUL
+                break;
+            }
+
+            fclose(mountsFile);
+        }
+
+        tracingDir = staticTracingDirBuffer;
+        __atomic_store_n(pStaticTracingDir, tracingDir, __ATOMIC_RELEASE); // CONSUME semantics.
```

Comment:
This is preferable over std atomis?

Reply:
How do you make an atomic variable that refers to a chunk of a memory-mapped buffer? I have a pointer to memory, not a variable of type `std::atomic<uint64_t>`.